crypto/stateproof/structs.go

```diff
@@ -82,6 +82,10 @@ type StateProof struct {
 	PositionsToReveal []uint64          `codec:"pr,allocbound=MaxReveals"`
 }
 
+// SigPartProofMaxSize is the maximum valid size of SigProofs and PartProofs elements of the Stateproof struct in bytes.
+// It is equal to merklearray.ProofMaxSizeByElements(config.StateProofTopVoters/2)
+const SigPartProofMaxSize = 35353
```

Comment:
agree, need to unify

Reply:
To me the cases seem different enough to warrant leaving separate.

SingleLeafProof is a valid type and right now it's only used in `merklesignature.Signature` but if it were used elsewhere the code would either:
 - stop compiling if we kept `//msgp:ignore maxsize SingleLeafProof` 
 - if we removed it, it would compile and give absurdly large values by calculating the size of the whole Proof with no limit to the number of elements. 

SigProof and PartProof on the other hands are just fields and not special type so the constant value totalallocbound was the only way of limiting their size.

Reply:
Some more thoughts:
We can't add `totalallocbound/maxtotalbytes` to `SingleLeafProof` directly at definition since `Proof` is an embedded field and current msgp implementation doesn't accept annotation on those. 

The `ProofMaxSizeByElements` function has 3 distinct inputs:

- Max size matching full potential `Proof`(unused)
- SigProof and PartProof size depending on the StateProofTopVoters constant
- SingleLeafProof which is dependent on the MaxEncodedTreeDepth